apps/functions-subscription/api/internal.yaml

```diff
@@ -0,0 +1,132 @@
+openapi: 3.0.1
+info:
+  version: 0.0.0
+  title: Trial System API
+  description: |
+    Documentation of the Trial System Subscription function.
+servers:
+  - url: https://ts-p-itn-subscription-fn-01.azurewebsites.net
+security:
+  - ApiKeyAuth: []
+paths:
+  /info:
+    get:
+      operationId: info
+      responses:
+        '200':
+          $ref: '#/components/responses/200Info'
```

Comment:
Please note that changing this component affects autogenerated model name. So we have to refactor the handler too.

Reply:
The generated definition has the same structure, so there is no need to refactor the handler. There is an error using the following options: `--request-types`, `--response-decoders` and `--client`. For all of them the error is:

```
ProblemJson -> src/generated/definitions/internal/ProblemJson.ts
ApplicationInfo -> src/generated/definitions/internal/ApplicationInfo.ts
CreatedAt -> src/generated/definitions/internal/CreatedAt.ts
UpdatedAt -> src/generated/definitions/internal/UpdatedAt.ts
UserId -> src/generated/definitions/internal/UserId.ts
SubscriptionState -> src/generated/definitions/internal/SubscriptionState.ts
TrialId -> src/generated/definitions/internal/TrialId.ts
CreateSubscription -> src/generated/definitions/internal/CreateSubscription.ts
Subscription -> src/generated/definitions/internal/Subscription.ts
Error: Error: Not implemented
npm ERR! Lifecycle script `generate` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: functions-subscription@0.0.0 
npm ERR!   at location: /.../trial-system/apps/functions-subscription
```

Reply:
UPDATE:
The generator raise an error due to a [reference on a parameter](https://github.com/pagopa/trial-system/pull/17/files#diff-88c7886cb4ef65b0e085738aa366872a7414912537c998672ef34a8e1f9e86f2R79-R87). The client is generated without error after removing the reference, but the type of the generated response in `unknown`. I suggest to keep the structure as is, and try to figure out why the generator does not handle it correctly.